src/aggregated/circulating_supply/circulating_supply_provider.rs

```diff
@@ -0,0 +1,180 @@
+use std::ops::{Add, Div};
+use std::str::FromStr;
+use std::time::{Duration, SystemTime};
+
+use actix::Addr;
+use actix_diesel::Database;
+use actix_web::rt::time;
+use bigdecimal::{BigDecimal, ToPrimitive};
+use diesel::PgConnection;
+use tokio::time::Instant;
+use tracing::{error, info};
+
+use near_indexer::near_primitives;
+
+use crate::aggregated::account_details;
+use crate::aggregated::account_details::get_account_balance;
+use crate::aggregated::circulating_supply::lockup;
+use crate::aggregated::circulating_supply::lockup::TRANSFERS_ENABLED;
+use crate::db_adapters::accounts;
+use crate::db_adapters::aggregated::circulating_supply::{
+    add_circulating_supply, get_precomputed_circulating_supply_for_timestamp,
+};
+use crate::db_adapters::blocks::get_latest_block_before_timestamp;
+use crate::models::aggregated::circulating_supply::CirculatingSupply;
+use crate::models::Block;
```

Comment:
We prefer to `use crate::models` and then use `models::Block` and such to communicate that the `Block` type is the database model rather than nearcore `Block` or any other block.